src/doc/src/reference/unstable.md

```diff
@@ -199,6 +199,36 @@ cargo +nightly build -Z config-profile
 ```
 
 
+### Build Profile
+* Tracking Issue: [rust-lang/rust#48683](https://github.com/rust-lang/rust/issues/48683)
+
+The `build` profile controls the build settings for build scripts,
+proc-macros, compiler plugins, and all of their dependencies. It requires the
+`build-profile` feature to be enabled.
+
+```toml
+cargo-features = ["build-profile"]
+
+[profile.build]
+# The following illustrates the defaults.
+opt-level = 0
+debug = false
+rpath = false
+lto = false
+debug-assertions = false
+codegen-units = 16
+panic = 'unwind'
+incremental = false
```

Comment:
Shouldn't this be `true`? (was this copy/pasted from somewhere else that needs an update?)

Reply:
Right now it is `false`.  The default build profile is defined [here](https://github.com/rust-lang/cargo/blob/59139bcbd7f654aaa697165f739793eea91ea51a/src/cargo/core/profiles.rs#L560) based on the default [here](https://github.com/rust-lang/cargo/blob/59139bcbd7f654aaa697165f739793eea91ea51a/src/cargo/core/profiles.rs#L464). This is similar to `release` mode.

I don't have a strong opinion about any of the defaults.  I think the theory on this one is that build scripts are rarely modified.  But I can see how it would be annoying when you are actively working on it.

Maybe you are thinking of #6564 which hasn't merged, yet?  That would change the default.

Reply:
Oh nah I just wanted to confirm. I think that we should have incremental turned on by default as the overhead only applies for `path` dependencies anyway, in this case typically just the build script itself. 

I could see this going either way though. Build scripts are typically quite small and fast to compile, which means that incremental isn't a hit for them nor does it really matter too much. I'd personally err on the side of enabling incremental though